cmd/aries-agentd/startcmd/start_test.go

```diff
@@ -0,0 +1,344 @@
+	
+	require.NotNil(t, flag)
+	require.Equal(t, flagName, flag.Name)
+	require.Equal(t, flagShorthand, flag.Shorthand)
+	require.Equal(t, flagUsage, flag.Usage)
+	require.Equal(t, "", flag.Value.String())
+
+	flagAnnotations := flag.Annotations
+	require.Len(t, flagAnnotations, 1)
+	requiredFlagKeyName := "cobra_annotation_bash_completion_one_required_flag"
+	require.Contains(t, flagAnnotations, requiredFlagKeyName)
+	requiredFlagAnnotation := flagAnnotations[requiredFlagKeyName]
+	require.Len(t, requiredFlagAnnotation, 1)
+	require.Equal(t, requiredFlagAnnotation[0], "true")
+
+	return
+}
+
+func TestStartAriesDRequests(t *testing.T) {
+	// TODO - remove this path manipulation after implementing #175 and #148
+	path, cleanup := generateTempDir(t)
+	defer cleanup()
+
+	testHostURL := randomURL()
+	testInboundHostURL := randomURL()
+	go startAgent(&RealServer{}, testHostURL, testInboundHostURL, path)
+	// give some time for server to start
+	if err := listenFor(testHostURL, time.Second); err != nil {
+		t.Fatal(err)
+	}
+	if err := listenFor(testInboundHostURL, time.Second); err != nil {
+		t.Fatal(err)
+	}
+	validateRequests(t, testHostURL, testInboundHostURL)
+}
+
+func listenFor(host string, d time.Duration) error {
+	timeout := time.After(d)
+	for {
+		select {
+		case <-timeout:
+			return errors.New("timeout: server is not available")
+		default:
+			conn, err := net.Dial("tcp", host)
+			if err != nil {
+				continue
+			}
+			if err := conn.Close(); err != nil {
+				return err
+			}
+			return nil
+		}
+	}
+}
+
+// TODO: this method should be refactored (e.g., too many lines).
+//nolint:funlen
+func validateRequests(t *testing.T, testHostURL string, testInboundHostURL string) {
+	newreq := func(method, url string, body io.Reader, contentType string) *http.Request {
+		r, err := http.NewRequest(method, url, body)
+		if contentType != "" {
+			r.Header.Add("Content-Type", contentType)
+		}
+		if err != nil {
+			t.Fatal(err)
+		}
+		return r
+	}
+
+	tests := []struct {
+		name               string
+		r                  *http.Request
+		expectedStatus     int
+		expectResponseData bool
+	}{
+		// controller API test
+		{
+			name:               "1: testing get",
+			r:                  newreq("GET", fmt.Sprintf("http://%s/connections/create-invitation", testHostURL), nil, ""),
+			expectedStatus:     http.StatusOK,
+			expectResponseData: true,
+		},
+
+		// DIDComm inbound API test
+		{
+			name: "200: testing didcomm inbound",
+			r: newreq("POST",
+				fmt.Sprintf("http://%s", testInboundHostURL),
+				strings.NewReader(`
+							{
+								"@id": "5678876542345",
+								"@type": "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/didexchange/1.0/invitation"
+							}`),
+				"application/didcomm-envelope-enc"),
+			expectedStatus:     http.StatusInternalServerError,
+			expectResponseData: false,
+		},
+	}
+
+	for _, tt := range tests {
+
+		resp, err := http.DefaultClient.Do(tt.r)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		defer func() {
```

Comment:
Do not use defer in a loop it can be a potential memory leak

Reply:
This wasn't actually code that I wrote - I just moved it from the previous file to here. Can you make a separate issue for this? Or if you know how to refactor the method then let me know and I can implement it if it doesn't take too long.